lib/rester/client.rb

```diff
@@ -31,6 +31,16 @@ def request(verb, path, params={}, &block)
       _process_response(path, *adapter.request(verb, path, params, &block))
     end
 
+    def with_context(context, &block)
+      unless adapter.is_a?(Adapters::StubAdapter)
+        raise Errors::MethodError, 'Can only use "with_context" with a StubAdapter'
+      end
+
+      adapter.context = context
+      yield block
+      adapter.context = nil
+    end
+
```

Comment:
Can this be done with a method_missing? I want to avoid having adapter specific logic in the client.

Could be something like:
```ruby
def method_missing(meth, *args, &block)
  if adapter.respond_to?(meth)
    adapter.public_send(meth, *args, &block)
  else
    super
  end
end
```